include/zephyr/dt-bindings/pinctrl/nrf-pinctrl.h

```diff
@@ -191,7 +201,19 @@
  * @param pin Pin (0..31).
  */
 #define NRF_PSEL(fun, port, pin)						       \
-	((((((port) * 32U) + (pin)) & NRF_PIN_MSK) << NRF_PIN_POS) |	       \
+	((((((port) * 32U) + (pin)) & NRF_PIN_MSK) << NRF_PIN_POS) |		       \
+	 ((NRF_FUN_ ## fun & NRF_FUN_MSK) << NRF_FUN_POS))
+
+/**
+ * @brief Utility macro to build nRF psels property entry when a pin is disconnected.
+ *
+ * This can be useful in situations where code running before Zephyr, e.g. a bootloader
+ * configures pins that later needs to be disconnected.
+ *
+ * @param fun Pin function configuration (see NRF_FUN_{name} macros).
+ */
+#define NRF_PSEL_DISCONNECTED(fun)						       \
+	(NRF_PIN_DISCONNECTED |							       \
```

Comment:
For consistency, it would be better to have `(NRF_PIN_DISCONNECTED << NRF_PIN_POS)` here. Just in case the order of bit fields in the value was to change at some point.